designdocs/batched_messages.md

```diff
@@ -0,0 +1,81 @@
+# Batched Messages
+
+## Table of Contents
+* [Context and Scope](#context-and-scope)
+* [Problem Statement](#problem-statement)
+* [Goals and Non-Goals](#goals-and-non-goals)
+* [Proposal](#proposal)
+* [Benefits and Risks](#benefits-and-risks)
+* [Alternatives and Rationale](#alternatives-and-rationale)
+* [Glossary](#glossary)
+
+## Context and Scope
+This design document describes batched messages and APIs for sending and retrieving them.
+
+## Problem Statement
+In order to reduce costs for announcers of messages on-chain as well as reduce network congestion, announcers collate messages into batches of the same type of message and announce the batch location on-chain, instead of announcing each individual message.
+
+## Goals and Non-Goals
+This specifies how batches are to be announced on chain: what is required and how a batch may be partially verified based on on-chain information.
+
+This document does not describe the messages that are supported for batch announcements. Please see the [DSNP Specification](https://spec.dsnp.org/).
+
+This document does not discuss validation of batch file format located at the URL in the announcement, since
+the file format cannot be verified on-chain. For details about batch file format, see [DSNP Spec: Batch Publications](https://spec.dsnp.org/DSNP/BatchPublications).
+
+## Proposal
+* All names are placeholders and may be changed.
+* Errors in the extrinsic(s) must have different, reasonably-named error enums for each type of error for ease of debugging.
+
+### Constants
+* `BatchPageSizeMax` - the maximum number of batch announcements that will be returned by the `get_batches` query
+* `BatchSizeMinBytes` - the minimum possible size of a valid batch file with 1 row
+
+### Enums
+* `MessageType` - all values specified in [DSNP Spec Announcement Types](https://spec.dsnp.org/DSNP/Announcements.html)
+
+### Types
+* `BatchAnnouncementParams<T:Config>`: generic
+    * `batch_url`:`Vec` the URL of the batch file.  Must be IPFS or HTTPS URL. The URL must be correctly formatted.
```

Comment:
This was a `&str`, @enddynayn suggested it be a `Vec` and now I wonder if it should be a `BoundedVec`, and if so what the bounds would be.